### PR TITLE
made questionnaire service, model, renamed api to database, added cli…

### LINKED ISSUE #60: Questionnaire service and schema

### PR DESCRIPTION
**Pull request by McDuuck**
…ent folder and index.ts, added serverless resource and updated readme

### CODE REVIEW

File: src/database/schemas/questionnaire/questionnaire.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  title: string;
+  description: string;
+  options: Question[];
+  tags?: string[];
+  passedUsers?: number[];
+  passScore: number;
+}
+
+export interface Question {
+  question: string;
```
Comment (dalju005-xamk) [inline src/database/schemas/questionnaire/questionnaire.ts:18]: Could this be question -> questionText, and in options value could be isCorrect ?
Do we need interface to passedUsers? Or could all that information be in one object ?
Could passScore be more like passingScore or minPassScore ?

**Review by dalju005-xamk — APPROVED** [review]

File: serverless.ts [hunk 2]
```diff
@@ -159,6 +159,23 @@ const serverlessConfiguration: AWS = {
       concurrency: 10,
     },
   },
+  resources: {
+    Resources: {
+      QuestionnaireTable: {
```
Comment (Jdallos) [inline serverless.ts]:
```suggestion
      Questionnaires: {
```

File: serverless.ts [hunk 3]
```diff
@@ -159,6 +159,23 @@ const serverlessConfiguration: AWS = {
...
+        Type: "AWS::DynamoDB::Table",
+        DeletionPolicy: "Delete",
+        Properties: {
+          TableName: "questionnaire",
```
Comment (Jdallos) [inline serverless.ts]:
```suggestion
          TableName: "questionnaires",
```

File: src/database/schemas/questionnaire/questionnaire.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+export interface QuestionnaireModel {
```
Comment (Jdallos) [inline src/database/schemas/questionnaire/questionnaire.ts:4]: Add Docs

File: src/database/schemas/questionnaire/questionnaire.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  id: number;
+  title: string;
+  description: string;
+  options: Question[];
+  tags?: string[];
+  passedUsers?: number[];
+  passScore: number;
+}
+
+export interface Question {
```
Comment (Jdallos) [inline src/database/schemas/questionnaire/questionnaire.ts:17]: Add Docs

File: src/database/schemas/questionnaire/questionnaire.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  question: string;
+  options: Option[];
+}
+
+export interface Option {
```
Comment (Jdallos) [inline src/database/schemas/questionnaire/questionnaire.ts:25]: Add Docs

File: src/database/services/quiz-api-service.ts [hunk 7]
```diff
@@ -0,0 +1,102 @@
+import { DocumentClient } from "aws-sdk/clients/dynamodb";
+import { QuestionnaireModel } from "../schemas/questionnaire/questionnaire";
+
+const tableName = process.env.QUESTIONNAIRE_TABLE;
```
Comment (Jdallos) [inline src/database/services/quiz-api-service.ts]: Looking at the elastic app functions I don't think the table name needs to be an env. Lets remove it as an env (as we name it in the serverless.ts anyway) and just add it here as a string

Comment (Jdallos) [inline src/database/services/quiz-api-service.ts]: Lets treat this as a const so rename TABLE_NAME

**Review by Jdallos — CHANGES_REQUESTED** [review]

File: src/database/schemas/questionnaire/questionnaire.ts [hunk 8]
```diff
@@ -1,3 +1,7 @@
+/**
+ * DynamoDB schema for Questionnaire
+ */
+
```
Comment (Jdallos) [inline src/database/schemas/questionnaire/questionnaire.ts]:
```suggestion
```

File: src/database/schemas/questionnaire/questionnaire.ts [hunk 9]
```diff
@@ -8,11 +12,19 @@ export interface QuestionnaireModel {
   passScore: number;
 }
 
+/**
+ * DynamoDB schema for Question
+ */
+
```
Comment (Jdallos) [inline src/database/schemas/questionnaire/questionnaire.ts]:
```suggestion
```

File: src/database/schemas/questionnaire/questionnaire.ts [hunk 10]
```diff
@@ -8,11 +12,19 @@ export interface QuestionnaireModel {
...
 export interface Question {
   question: string;
   options: Option[];
 }
 
+/**
+ * DynamoDB schema for Option
+ */
+
```
Comment (Jdallos) [inline src/database/schemas/questionnaire/questionnaire.ts]:
```suggestion
```

**Review by Jdallos — APPROVED** [review]

**Pull request merged**